src/lib/shady-render.ts

```diff
@@ -107,53 +107,48 @@ const shadyRenderSet = new Set<string>();
  * not be scoped and the <style> will be left in the template and rendered
  * output.
  */
-const ensureStylesScoped =
+const styleTemplatesForScope =
     (fragment: DocumentFragment, template: Template, scopeName: string) => {
-      // only scope element template once per scope name
-      if (!shadyRenderSet.has(scopeName)) {
-        shadyRenderSet.add(scopeName);
-        const styleTemplate = document.createElement('template');
-        Array.from(fragment.querySelectorAll('style')).forEach((s: Element) => {
-          styleTemplate.content.appendChild(s);
-        });
-        window.ShadyCSS.prepareTemplateStyles(styleTemplate, scopeName);
-        // Fix templates: note the expectation here is that the given `fragment`
-        // has been generated from the given `template` which contains
-        // the set of templates rendered into this scope.
-        // It is only from this set of initial templates from which styles
-        // will be scoped and removed.
-        removeStylesFromLitTemplates(scopeName);
-        // ApplyShim case
-        if (window.ShadyCSS.nativeShadow) {
-          const style = styleTemplate.content.querySelector('style');
-          if (style !== null) {
-            // Insert style into rendered fragment
-            fragment.insertBefore(style, fragment.firstChild);
-            // Insert into lit-template (for subsequent renders)
-            insertNodeIntoTemplate(
-                template,
-                style.cloneNode(true),
-                template.element.content.firstChild);
-          }
+      shadyRenderSet.add(scopeName);
+      // Move styles out of rendered DOM and store.
+      const styleFragment = document.createDocumentFragment();
+      Array.from(fragment.querySelectorAll('style')).forEach((s: Element) => {
+        styleFragment.appendChild(s);
+      });
+      // Remove styles from nested templates in this scope and put them
+      // into the "root" template passed in as `template`.
+      removeStylesFromLitTemplates(scopeName);
+      insertNodeIntoTemplate(
```

Comment:
Is this the big change here? I'm having a little bit of a hard time seeing the change in behavior.

Reply:
Added a comment. Yes, the key bit is that we hand `ShadyCSS` a template that will actually be rendered by `lit-html`. This way it can update the style inside it and affect subsequent `lit-html` renderings.